### PR TITLE
feat: Fixes always idle and adds noWorkSpaceText

### LINKED ISSUE #137: presence always showing idle
Suddenly rich presence only shows idle even if I am currently coding.
image

### PR DESCRIPTION
**Pull request by nick22985**
Should fix #137

### CODE REVIEW

File: src/activity.ts [hunk 1]
```diff
@@ -105,14 +105,15 @@ export const activity = async (
         (isExcluded(config.get(CONFIG_KEYS.Ignore.Workspaces)!, dataClass.workspaceFolder.uri.fsPath) ||
             isExcluded(config.get(CONFIG_KEYS.Ignore.Workspaces)!, dataClass.workspaceName));
 
-    const isNotInWorkspace = !isWorkspaceExcluded && (!dataClass.workspaceFolder || !dataClass.editor);
+    const isNotInWorkspace = (!isWorkspaceExcluded && (!dataClass.workspaceFolder));
+    const isNotInFile = (!isWorkspaceExcluded && !dataClass.editor);
```
Comment (nick22985) [inline src/activity.ts:109]: Added this if we do future things needing to tell if that it is not in a file / in a workspace independently

**Comment by xhayper** [comment]
will require more stuff to be add, example being icon and statue and button and all those funny stuff

i will take it from here

**Comment by nick22985** [comment]
Cool cool didn't really do any of the customisation stuff just a basic fix for the problem ty

File: src/activity.ts [hunk 2]
```diff
@@ -348,10 +349,11 @@ export const replaceFileInfo = async (
     const config = getConfig();
     text = text.slice();
 
-    let workspaceFolderName = dataClass.workspaceFolder?.name ?? FAKE_EMPTY;
-    let workspaceName = dataClass.workspaceName ?? FAKE_EMPTY;
+    let workspaceFolderName = dataClass.workspaceFolder?.name ?? config.get(CONFIG_KEYS.Status.Details.Text.noWorkSpaceText);
+    let workspaceName = dataClass.workspaceName ?? config.get(CONFIG_KEYS.Status.Details.Text.noWorkSpaceText);
     let workspaceAndFolder =
         workspaceName + (workspaceFolderName != FAKE_EMPTY ? ` - ${workspaceFolderName}` : FAKE_EMPTY);
+    workspaceAndFolder == '' ? config.get(CONFIG_KEYS.Status.Details.Text.noWorkSpaceText) : workspaceAndFolder;
```
Comment (nick22985) [inline src/activity.ts:356]: Also this doesnt work just noticed it

**Comment by nick22985** [comment]
one sec

**Comment by nick22985** [comment]
fixed ssh leaking lol

**Comment by nick22985** [comment]
Yep I am all done

**Pull request merged**